hw/opentitan/ot_spi_device.c

```diff
@@ -2033,12 +2044,32 @@ static void ot_spi_device_chr_receive(void *opaque, const uint8_t *buf,
         break;
     }
 
-    if (!bus->byte_count) {
-        if (bus->release) {
-            ot_spi_device_release(s);
-        } else {
-            BUS_CHANGE_STATE(s, IDLE);
-        }
+    if (bus->byte_count) {
+        /* there is more to process in this state */
+        return;
+    }
+
+    if (bus->release) {
+        /* final packet in transaction, release CS */
+        ot_spi_device_release(s);
+        return;
+    }
+
+    /* more packets to process in transaction, prepare to receive another header
+     */
```

Comment:
nit pick, QEMU comment style
```suggestion
    /* 
     * more packets to process in transaction, prepare to receive another header
     */
```